format/swf/lite/MorphShape.hx

```diff
@@ -84,31 +91,54 @@ class MorphShape extends Shape {
 
 	public override function __update (transformOnly:Bool, updateChildren:Bool):Void {
 
-		if(__renderDirty){
+        __updateTransforms();
 
-			var twips_ratio = Math.floor(ratio*65535);
-			var handler = __symbol.cachedHandlers.get(twips_ratio);
-			if ( handler == null ) {
-				var swf_shape = __symbol.getShape(ratio);
+		if(__renderDirty){
+            var cacheEntry = __symbol.getCacheEntry(__renderTransform, ratio);
+
+            if(cacheEntry == null) {
+                var twips_ratio = Math.floor(ratio*65535);
+                var handler = __symbol.cachedHandlers.get(twips_ratio);
+                if ( handler == null ) {
+                    var swf_shape = __symbol.getShape(ratio);
+
+                    handler = new ShapeCommandExporter ();
+                    swf_shape.export (handler);
+                    __symbol.cachedHandlers.set(twips_ratio, handler);
+                }
+
+                graphics.clear();
+                ShapeSymbol.processCommands(graphics, handler.commands);
+
+                #if(profile && js)
+                    var profileId = getProfileId();
+                    var value = __updateCount.get(profileId);
+                    value = value != null ? value : 0;
+                    __updateCount.set(profileId, value + 1);
+                #end
+
+                mustCache = true;
+
+            } else {
+                Reflect.setField(graphics, "__bitmap", cacheEntry.bitmapData);
+                Reflect.setField(graphics, "__dirty", false);
+                @:privateAccess graphics.__bounds = cacheEntry.bounds;
+            }
+	    }
 
-				handler = new ShapeCommandExporter ();
-				swf_shape.export (handler);
-				__symbol.cachedHandlers.set(twips_ratio, handler);
-			}
+		super.__update(transformOnly, updateChildren);
```

Comment:
I know it was like this before your changes, but can't we call super.__update() before processing the morph shape? That would save the first __updateTransforms() call

Reply:
+ tabs